nixos/modules/services/web-apps/trilium.nix

```diff
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.trilium-server;
+in
+{
+
+  options.services.trilium-server = with lib; {
+    enable = mkEnableOption "trilium-server";
+
+    dataDir = mkOption {
```

Comment:
Does this option provide value to you? Do you have a use case for this option where you wouldn't use the default you might share?

Reply:
We have already discussed this once on this [PR](https://github.com/NixOS/nixpkgs/pull/60788#issuecomment-491436203).

Bottom line: I like my system to not break because some default changes without me noticing.
If the Postgres socket directory had been a configuration option instead of a magic value, changing it from `/tmp` to `/run/postgres` would not have required a backwards incompatible change with 19.09 because one could just reference it as `${services.porstgres.socketDir}` in depending services.
With data, I think it's vital that this is controllable because backups will just silently continue to run on the old paths without braking, for example. Same is true for mount points, quotas, (sub-) volumes, etc.

Reply:
Right. Thanks for reminding me. We continue to respectfully disagree on this topic, but as I'll continue to mention I'm happy to accept modules which include a `dataDir` option.

Reply:
I have also suggested databases have a `socket` option, but `readOnly`.

Reply:
> I have also suggested databases have a `socket` option, but `readOnly`.

I would be very much in favor of this!